lib/extract-js.js

```diff
@@ -0,0 +1,216 @@
+var htmlparser = require('htmlparser2');
+var Errors = require('./errors');
+
+/**
+ * Html file representation (needs for errors output).
+ *
+ * @name HtmlFile
+ */
+var HtmlFile = function(filename, source) {
+    this._filename = filename;
+    this._lines = source.split(/\r\n|\r|\n/);
```

Comment:
`\r` isn't used on Mac since ages :wink: To deal with files, where mixed line endings exists I usually do this:

1. replace `\r\n` with `\n`
2. replace `\r` with `\n`
3. split by `\n`